torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -16,24 +16,33 @@
 import numpy as np
 import functools
 import itertools
-from typing import Tuple, Union, List, Sequence, Any, Optional, Set
+from typing import Union, Sequence, Any, Optional
 from enum import IntEnum
 
 from torch.amp import custom_fwd, custom_bwd
 
+PartitionSpec = tuple[Union[tuple[Union[int, str], ...], int, str, None], ...]
+"""PartitionSpec describes the sharding of a tensor.
+
+Specifically, it is a tuple where each element references one or more device mesh
+axes, and instructs to shard the corresponding dimension of the input tensor over
+those mesh axes.
+"""
+
 
 class Mesh:
   """Describe the logical XLA device topology mesh and the underlying resources.
 
   Args:
-    device_ids (Union[np.ndarray, List]): A raveled list of devices (IDs) in a custom order. The list is reshaped
-        to an `mesh_shape` array, filling the elements using C-like index order.
+    device_ids: A raveled list of devices (IDs) in a custom order.
+        The list is reshaped to an `mesh_shape` array, filling the elements using C-like index order.
+        Each ID indexes into the list of devices returned by `xr.global_runtime_device_attributes()`.
 
-    mesh_shape (Tuple[int, ...]): A int tuple describing the logical topology shape
+    mesh_shape: A int tuple describing the logical topology shape
         of the device mesh, and each element describes the number of devices in
         the corresponding axis.
 
-    axis_names (Tuple[str, ...]): A sequence of resource axis names to be assigned to the dimensions
+    axis_names: A sequence of resource axis names to be assigned to the dimensions
```

Comment:
A sequence of axis names, each name is assigned to the corresponding dimension of the `devices` argument.

Reply:
done